MGDOBR-479: replicate sso.redhat.com jwt claims

r00ta opened the pull request:
[MGDOBR-479](https://issues.redhat.com/browse/MGDOBR-479)

This PR aims to add the hardcoded service accounts for the user kermit, the shard and the webhook robot. In addition to that, claims are injected into the jwt tokens for users according to the sso.redhat.com structure 

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.

</details>

robpblake commented on kustomize/overlays/stable/manager/keycloak-realm.yaml:
```diff
@@ -7,7 +7,7 @@ metadata:
 spec:
   encryptedData:
-    event-bridge-fm-realm-sample.json: AgBBRGZj9vPdbBm3b1AijWswtLi2vVPunTqh1wih0gIO6kENEME+bgyCLxpnYKTInhSdjmkzoPsWorZFJxNkHlxWZVAGx0UMylssC1Kg31WOINsR+fLGhkesiXC/GQUmZ4mfhS5Svrw36EXOGm0RR0Q/boGdDdIlmYsL+0vSlQrZg48RsIv86bVi2x0XtMM43kc3VAopx70lLbR4c3P/6zYr2J8gEiA1mPw6jvWES+vROTQW0KoPbxsrmvJAd8VKZOZ9t6tvABdlFA7uZ8pry5W7w1aUa+HTnWtU70c089zZgCBR/Vq4HaFGrfkRbaajyES3mMMK1NAG/a38zibEkLlF5pxKI2g+i6gjz5GG ...
+    event-bridge-fm-realm-sample.json: AgBmjShnK+w7/6qjkhwNZsiSPCyLTdoe/gEiips5N3Hq8XUUGDyuOZMmDDcbayfsqkXhvF9zlL+8N0qIMJ8GQE8tUFadFeL2btpf9OT0r0qWXpAFheMR0WHRdB4GQ2Xh8stHF0xo/QzCs9GP31rB9O2KqaC8ZOcYtwIIR1W1DGaF34onb2zNinQTLLqd9fFf/H7rxVkwHhMubgM8F1n7jKasb1+k6juyhbge5uYZNS6D9KZ1Opv0rWbJj8Ov5b5Ku41jH7y1f/gYga5yaoTmvIkHXio9JBRZRbZ2iqSKJs0nvNTS8OD0jDWmSu9m0G1DEzJZgPnq9j2P0EKe02VXPN9E5yi3oAdXFujP5lMz ...
```
@r00ta Should we do stable in other commit once we're happy this works in `dev`?

r00ta replied:
done

robpblake reviewed (commented):
@r00ta We need to remember to roll the Fleet Shard Operator pod when this one is merged. Last time we updated the Keycloak configuration I seem to remember that caused us problems with the Shard being unable to refresh it's access token.

My other question is around separating out the deployment of these changes into our `stable` environment.  We don't have a defined process yet for when we promote changes from `dev` to `stable`. However we must verify that `stable` is working after we make a change to it.

This PR raises a good question of when we deploy environment configuration changes into `stable`. I think we should treat that in the same way as we treat a code deployment.

WDYT?

cc @danielezonca @lucamolteni @kostola

r00ta commented:
> @r00ta We need to remember to roll the Fleet Shard Operator pod when this one is merged. Last time we updated the Keycloak configuration I seem to remember that caused us problems with the Shard being unable to refresh it's access token.
> 
> My other question is around separating out the deployment of these changes into our `stable` environment. We don't have a defined process yet for when we promote changes from `dev` to `stable`. However we must verify that `stable` is working after we make a change to it.
> 
> This PR raises a good question of when we deploy environment configuration changes into `stable`. I think we should treat that in the same way as we treat a code deployment.
> 
> WDYT?
> 
> cc @danielezonca @lucamolteni @kostola @manstis @vaibhavjainwiz @msmagnanijr

Reverted the changes in stable. Can you approve if you are ok with the changes?
